src/routes/pages.ts

```diff
@@ -145,6 +145,24 @@ function setRoutes(redisClient: redis.RedisClient): express.Router {
         const mainGenresMap: Map<string, StatefulObject>     = new Map<string, StatefulObject>();
         const specificGenresMap: Map<string, StatefulObject> = new Map<string, StatefulObject>();
         const daysMap: Map<string, StatefulObject>           = new Map<string, StatefulObject>();
+
+        const dayNumbers: number[] = await redisHelper.getLineupDays(redisClient, festival.name, queryYear);
+
+        const daysWithMetadata: LineupDay[] = await Promise.all(dayNumbers.map(
+            async (day) => redisHelper.getLineupDayMetadata(redisClient, festival.name, queryYear, day)));
+
+        for (const lineupDay of daysWithMetadata) {
```

Comment:
I'm not totally sure if `daysWithMetadata` can be null or if it will always be a list returned from Promise.all, but even if it's a list it's possible to be a list of nulls, which will error as soon as `lineupDay` is accessed. The two redis functions `getLineupDays` and `getLineupDayMetadata` should either be changed to always return an empty list if the final if-check fails in the function (instead of just not returning anything) or everything that calls those functions needs to be null-checked, so checking `dayNumbers` before dayNumbers.map is called and somehow checking the contents of what's returned by Promise.all